### PR TITLE
defect #2401179: fix bdd scenario file name

### PR DESCRIPTION
**Pull request by pikachugb**
https://center.almoctane.com/ui/entity-navigation?p=1001/1002&entityType=work_item&id=2401179

**Problem:** When downloading the script for BDD Scenario entity, before it was "testName_testId.feature", now it should be "bddSpecTestName_bddSpecTestId.feature", where the bdd spec si the parent of the scenario.

**Solution:** 
- I added a new optional parameter to the method that retrieves all the fields of an entity, so that we can retrieve only the ones that we need (bdd_spec in our case)
- In the download-test file, I changed the logic for the case when we are dealing with BDD Scenario

### CODE REVIEW

File: src/octane/service/octane-service.ts [hunk 1]
```diff
@@ -489,16 +489,21 @@ export class OctaneService {
...
         const octaneType = subType || type;
         if (!octaneType) {
             return;
         }
-        const fields = await this.getFieldsForType(octaneType);
+
         if (!fields) {
-            this.logger.error(`Could not determine fields for type ${octaneType}.`);
-            return;
+            fields = await this.getFieldsForType(octaneType);
```
Comment (vladbenchea) [inline src/octane/service/octane-service.ts:499]: why did you move this line of code? also you removed const from it, it's used as a constant.

**Review by vladbenchea — APPROVED** [review]

**Review by Rogojan — APPROVED** [review]

**Review by RalucaOT — APPROVED** [review]
